src/interfaces/earnings.ts

```diff
@@ -227,32 +223,32 @@ export class EarningsInterface<C extends ChainId> extends ServiceInterface<C> {
       .map(day => blockNumber - day * this.blocksPerDay());
 
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const response = await this.yearn.services.subgraph.fetchQuery<any>(ASSET_HISTORIC_EARNINGS(blocks), {
+    const { data } = await this.yearn.services.subgraph.fetchQuery<any>(ASSET_HISTORIC_EARNINGS(blocks), {
       id: vault
     });
 
-    const data = response.data;
-
     const labels = blocks.map(block => `block_${block}`);
 
-    const token = data.vault.token.id as string;
+    const token = data.vault.token.id as Address;
     const priceUsdc = await this.yearn.services.oracle.getPriceUsdc(token).then(price => new BigNumber(price));
 
     const earningsDayData = labels.map(label => {
       const strategies: StrategiesResponse[] = data[label].strategies;
-      const totalGain = strategies
-        .map(strategy => (strategy.latestReport ? new BigNumber(strategy.latestReport.totalGain) : new BigNumber(0)))
-        .reduce((sum, value) => sum.plus(value));
 
-      const totalLoss = strategies
-        .map(strategy => (strategy.latestReport ? new BigNumber(strategy.latestReport.totalLoss) : new BigNumber(0)))
-        .reduce((sum, value) => sum.plus(value));
+      const { totalGain, totalLoss } = strategies.reduce(
+        ({ totalGain, totalLoss }, { latestReport }) => {
+          if (latestReport) {
+            totalGain = totalGain.plus(toBN(latestReport.totalGain));
+            totalLoss = totalLoss.plus(toBN(latestReport.totalLoss));
+          }
+          return { totalGain, totalLoss };
+        },
+        { totalGain: toBN(0), totalLoss: toBN(0) }
+      );
```

Comment:
Saves one loop through the strategies as `totalGain`/`totalLoss` have basically identical logic